src/plugins/ieo/components/OrderForm/index.tsx

```diff
@@ -0,0 +1,438 @@
+                this.props.listenInputPrice && this.props.listenInputPrice();
+                break;
+            case this.props.amountText:
+                this.setState(prev => ({
+                    amountFocused: !prev.amountFocused,
+                }));
+                break;
+            default:
+                break;
+        }
+    };
+
+    private handlePriceChange = (value: string) => {
+        const convertedValue = cleanPositiveFloatInput(String(value));
+        const condition = new RegExp(`^(?:[\\d-]*\\.?[\\d-]{0,${this.state.currentMarketBidPrecision}}|[\\d-]*\\.[\\d-])$`);
+        if (convertedValue.match(condition)) {
+            this.setState({
+                price: convertedValue,
+            });
+        }
+        this.props.listenInputPrice && this.props.listenInputPrice();
+    };
+
+    private handleAmountChange = (value: string) => {
+        const convertedValue = cleanPositiveFloatInput(String(value));
+        const condition = new RegExp(`^(?:[\\d-]*\\.?[\\d-]{0,${this.state.currentMarketAskPrecision}}|[\\d-]*\\.[\\d-])$`);
+        if (convertedValue.match(condition)) {
+            this.setState({
+                amount: convertedValue,
+            });
+        }
+    };
+
+    private handleChangeAmountByButton = (value: number, type: string) => {
+        switch (type) {
+            case 'buy':
+                switch (this.state.orderType) {
+                    case 'Limit':
+                        this.setState({
+                            amount: this.props.available && + this.state.price ? (
+                                Decimal.format(this.props.available / +this.state.price * value, this.state.currentMarketAskPrecision)
+                            ) : '',
+                        });
+                        break;
+                    case 'Market':
+                        this.setState({
+                            amount: this.props.available ? (
+                                Decimal.format(getAmount(Number(this.props.available), this.props.proposals, value), this.state.currentMarketAskPrecision)
+                            ) : '',
+                        });
+                        break;
+                    default:
+                        break;
+                }
+                break;
+            case 'sell':
+                this.setState({
+                    amount: this.props.available ? (
+                        Decimal.format(this.props.available * value, this.state.currentMarketAskPrecision)
+                    ) : '',
+                });
+                break;
+            default:
+                break;
+        }
+    };
+
+    private handleSubmit = () => {
+        const { available, type } = this.props;
+        const { amount, price, priceMarket, orderType } = this.state;
+
+        const order = {
+            type,
+            orderType,
+            amount,
+            price: orderType === 'Market' ? priceMarket : price,
```

Comment:
There is no ability in IEO to buy "Limit". Only "Market" and only "Buy" (orderType).
Please remove Limit and OrderType everywhere in this file (it`s dead code)